fix/workorder-detail-fix.lua

```diff
@@ -0,0 +1,199 @@
+    [df.job_type.SewImage] = true,
+    -- list may be incomplete
+}
+
+-- copy order.item fields/flags over to job's job_item
+-- only the essentials: stuff that is editable via gui/job-details
+local function correct_item_details(job_item, order_item)
+    local fields = {'item_type', 'item_subtype', 'mat_type', 'mat_index'}
+    for _, field in pairs(fields) do
+        job_item[field] = order_item[field]
+    end
+
+    local flags_names = {'flags1', 'flags2', 'flags3', 'flags4', 'flags5'}
+    for _, flags in pairs(flags_names) do
+        local order_flags = order_item[flags]
+        if type(order_flags) == 'number' then
+            job_item[flags] = order_flags
+        else
+            job_item[flags].whole = order_flags.whole
+        end
+    end
+end
+
+-- correct job's job_items to match the order's
+local function enforce_order_details(job, order)
+    if not job.job_items then return end -- never happens (error here?)
+    local modified = false
+    for idx, job_item in ipairs(job.job_items) do
+        local order_item = order.items[idx]
+        if not order_item then break end -- never happens (error here?)
+        if job_item.item_type ~= order_item.item_type then
+            -- dfhack's isSuitableItem function will allow the orders we want,
+            -- but disallow insane combinations like meals made of shoes
+            local suitable = dfhack.job.isSuitableItem(
+                job_item, order_item.item_type, order_item.item_subtype )
+            if suitable then
+                correct_item_details(job_item, order_item)
+                modified = true
+            else --[[ error on unsuitable item?]] end
+        end
+    end
+    if modified then jobs_corrected = jobs_corrected + 1 end
+end
+
+-- check orders list to see if fix is needed
+-- yields id-order table for the orders with issues, or nil when no issues
+local PrepareMeal = df.job_type.PrepareMeal
+local SewImage = df.job_type.SewImage
+local NONE = df.job_type.NONE
+local function get_bugged_orders()
+    local orders = {}
+    local num_bugged, improve_item_index, item = 0, 1, nil
+    for _, order in ipairs(df.global.world.manager_orders) do
+        if not order.items then goto nextorder end
+        local order_job_type = order.job_type
+        if not offending_jobs[order_job_type] then goto nextorder end
+        if #order.items == 0 then goto nextorder end -- doesn't happen
+
+        -- for PrepareMeal jobs, any one of the items could be an issue.
+        if order_job_type == PrepareMeal then
+            for _, _item in ipairs(order.items) do
+                if _item.item_type ~= NONE then goto fix end
+            end
+            goto nextorder
+        end
+
+        -- All other types are improve jobs; only the improved item is checked
+        -- Only SewImage has the item-to-improve at items[0]
+        improve_item_index = (order_job_type ~= SewImage) and 1 or 0
+        item = order.items[improve_item_index]
```

Comment:
missing `local` keywords

Reply:
had a weird time with `goto` jumping into locals' scope. Now they're inside a do/end block so `goto` can skip over it